include/deal.II/matrix_free/operators.h

```diff
@@ -160,6 +160,12 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
+     * Get read access to the MatrixFree object stored with this operator.
+     */
+    const std_cxx11::shared_ptr<const MatrixFree<dim,Number> > &
```

Comment:
It may be a bit cleaner to just return `const MatrixFree<dim, Number>` if all you intend to do is read the object.